app/controllers/api/vcs.php

```diff
@@ -593,32 +611,103 @@
         $files = \array_column($files, 'name');
         $languages = $github->listRepositoryLanguages($owner, $repositoryName);
 
-        $detectorFactory = new Detector($files, $languages);
+        $detector = new Packager($files);
+        $detector
+            ->addOption(new Yarn())
+            ->addOption(new PNPM())
+            ->addOption(new NPM());
+        $detectedPackager = $detector->detect();
+
+        $packagerName = !\is_null($detectedPackager) ? $detectedPackager->getName() : 'npm';
+
+        if ($type === 'framework') {
+            $detection = new Document([
+                'framework' => '',
+                'installCommand' => '',
+                'buildCommand' => '',
+                'outputDirectory' => '',
+            ]);
+
+            $frameworkDetector = new Framework($files, $packagerName);
+            $frameworkDetector
+                ->addOption(new Flutter())
+                ->addOption(new Nuxt())
+                ->addOption(new Astro())
+                ->addOption(new Remix())
+                ->addOption(new SvelteKit())
+                ->addOption(new NextJs());
+
+            $detectedFramework = $frameworkDetector->detect();
+
+            if ($detectedFramework) {
+                $framework = $detectedFramework->getName();
+                $detection->setAttribute('installCommand', $detectedFramework->getInstallCommand());
+                $detection->setAttribute('buildCommand', $detectedFramework->getBuildCommand());
+                $detection->setAttribute('outputDirectory', $detectedFramework->getOutputDirectory());
+            } else {
+                $framework = 'other';
+                $detection->setAttribute('installCommand', '');
+                $detection->setAttribute('buildCommand', '');
+                $detection->setAttribute('outputDirectory', './');
+            }
 
-        $detectorFactory
-            ->addDetector(new JavaScript())
-            ->addDetector(new Bun())
-            ->addDetector(new PHP())
-            ->addDetector(new Python())
-            ->addDetector(new Dart())
-            ->addDetector(new Swift())
-            ->addDetector(new Ruby())
-            ->addDetector(new Java())
-            ->addDetector(new CPP())
-            ->addDetector(new Deno())
-            ->addDetector(new Dotnet());
+            $frameworks = Config::getParam('frameworks');
+            $frameworkDetail = \array_reverse(\array_filter(\array_keys($frameworks), function ($key) use ($framework, $frameworks) {
+                return $frameworks[$key]['key'] === $framework;
+            }))[0] ?? '';
+            $detection->setAttribute('framework', $frameworkDetail);
```

Comment:
Can you please share with me what value of such $frameworkDetail is?  If it's just string, let's explore if we can simplify this code, I am almost certain it can be done.